agent/engine/task_manager.go

```diff
@@ -421,6 +423,9 @@ func (mtask *managedTask) handleContainerChange(containerChange dockerContainerC
 	// Container has progressed its status if we reach here. Make sure to save it to database.
 	defer mtask.engine.saveContainerData(container)
 
+	// if this is an execute command enabled STOPPED container event we should emit a managedAgent event
+	mtask.handleManagedAgentStoppedTransition(event.Status, container, execcmd.ExecuteCommandAgentName)
```

Comment:
can we call this after checking if container status is stopped and exec enabled ? would prefer not logging if container is not stopped each time.

Reply:
Agreed. Changed.

Reply:
The idea of a generic handleManagedAgentStoppedTransition is that all managed agents will likely have some stop container handling required; it would be nice to be able to say `for agent in agents: handleManagedAgentStoppedTransition(agent)`.

I was hoping to insulate agent specific conditional logic to the function itself.